snappy/info.go

```diff
@@ -162,6 +155,22 @@ func FindSnapsByNameAndVersion(needle, version string, haystack []*Snap) []*Snap
 	return found
 }
 
+// FindSnapsByNameAndRevision returns the snaps with the name/version in the
+// given slice of snaps
+func FindSnapsByNameAndRevision(needle string, revno int, haystack []*Snap) []*Snap {
```

Comment:
Can we please stick to "revision" here and everywhere else in the code?  It's slightly easier on the brain to use a single term consistently. It's okay for this branch to go in without modifying that, though, but when driving by code that has revno, I suggest switching over.

I'd also stick to more boring parameter names (name, revision, snaps), because otherwise the parameter list becomes unclear by itself. One needs to go look for what a "needle" is elsewhere.